paper/paper.md

```diff
@@ -0,0 +1,111 @@
+
+1. A workflow and method to annotate audio data.
+2. A function to prepare an audio dataset into train, test, and validation subsets.
+3. A function to train and evaluate a transformer model.
+4. A function to apply a trained model to a new audio file.
+
+## Related Work
+
+pyannote-audio
+
+transformers ?
+
+auto-ml?
+
+explosion ai ?
+
+## Workflow
+
+Speakerbox provides two methods to prepare a speaker identification training dataset: speaker diarization and labeling, and, importing annotations from other annotation tools.
+
+#### Diarization
+
+While diarization cannot be used as a single solution for multi-speaker multi-recording speaker identification, it can be used for quickly generating large amounts of training examples which can be validated and labeled for use in a later speaker identification training set.
+
+We make use of diarization as one method for preparing a speaker identification training dataset by using a diarization model provided by `pyannote.audio` to diarize an audio file and place the unlabeled chunks of audio into directories on the user's file system. A user can then listen to a few or all of the samples of audio in each directory, remove any samples that were mis-classified, and finally rename each of the directories with a speaker identifier (i.e. a name, database id, etc.).
+
+TODO WRITING: insert fig?
+
+#### Using Gecko Annotations
+
+To improve model accuracy and improve coverage of edge cases, users may find it useful to use Gecko: a free web application for manual segmentation of an audio files by speaker as well as annotation of the linguistic content of a conversation [@Gecko2019]. Speakerbox makes use of Gecko annotations as a method for training dataset creation by providing functions to split and prepare audio files using the annotations stored in a Gecko created JSON file.
+
+### Data Preparation
+
+Speakerbox is built with the goal of making multi-recording, multi-speaker identification model training as simple as possible while still attempting to help train an accurate, useful model for application. To ensure that the model is learning the features of each speaker's voice and not the features of the microphone or the specific words and phrases of the recording, we create dataset training, test, and evaluation splits based off of a recording holdout and speaker stratification pattern. Each train, test, or evaluation subset must contain unique recording IDs to reduce the chance of learning the features of specific microphones or recording contexts. We then check that our speaker stratifaction requirement is met by checking that each of the produced subsets contains recordings of every speaker available from the whole dataset. For example, if there are nine unique speakers in the dataset as a whole, then each train, test, and evaluation subset is required to have all nine speakers as well.
```

Comment:
The opening sentence is a useful explanation of SB goals -- think about using this earlier. And enumerate the goals so the reader can follow. 

Not relevant to this section, but a thought about the intro more generally:
- Start with a declarative sentence "Speakerbox is a python-based framework developed to improve X, Y and Z. It achieves this by A, B, and C which we explain in detail in the sections below.

Reply:
Hmmm can you expand on what you mean by the "enumerate the goals so the reader can follow". I agree this can be made more clear but I don't know how to enumerate the goals here.

Reply:
>Speakerbox is built with the goal of making multi-recording, multi-speaker identification model training as simple as possible while still attempting to help train an accurate, useful model for application. 

Speakerbox attempts to simply model training by making the following processes easier: 
- X
- Y 
- Z 

The goal of this package is to simplify these processes while maintaining fidelity in the training and utility of a model for application. (Say what kind of model)